demux/utils/samplesheet.py

```diff
@@ -206,6 +208,65 @@ def _validate_uniq_index(samplesheet):
         return True
 
 
+class MiSeqSamplesheet(Samplesheet):
+
+    def __init__(self, samplesheet_path):
+        self.samplesheet_path = samplesheet_path
+        converted_sheet_path = self.convert_sheet_type(samplesheet_path)
+        self.parse(converted_sheet_path)
+
+    def convert_sheet_type(self, samplesheet_path):
+        """
+        Conver miseq to hiseq style samplesheet for demultiplexing.
+        """
+        dirname = os.path.dirname(samplesheet_path)
+        flowcell_id = dirname.split(os.sep)[-1].split('_')[-1] 
+        deplex_file = os.path.join(dirname, 'demultiplex_sheet.csv')
+        header_line = "FCID,Lane,SampleID,SampleRef,Index,Description,Control,Recipe,Operator,SampleProject\n"
+        forbidden_name_chars = [' ','-', '/']
+
+        with open(samplesheet_path, 'r') as inputfile open(deplex_file 'w') as outputfile:
+            outputfile.write(header_line)
```

Comment:
Writing would happen in `demux/cli/samplesheet.py` (function `demux()`). We can sit together tomorrow to finish this if you like.